x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/template/IndexTemplateRegistry.java

```diff
@@ -842,8 +842,13 @@ public void onFailure(Exception e) {
 
     void onRolloversBulkResponse(Collection<RolloverResponse> rolloverResponses) {
         for (RolloverResponse rolloverResponse : rolloverResponses) {
-            if (rolloverResponse.isRolledOver() == false) {
-                logger.warn("rollover of the [{}] index [{}] failed", getOrigin(), rolloverResponse.getOldIndex());
+            if (rolloverResponse.isRolledOver() == false && rolloverResponse.isLazy() == false) {
+                logger.warn(
+                    "rollover of the [{}] index [{}] failed, rollover response was [{}]",
+                    getOrigin(),
+                    rolloverResponse.getOldIndex(),
+                    rolloverResponse
+                );
```

Comment:
Nit: should we make it more specific? If we pass here the `lazy` flag we can know what was expected in the response and log it accordingly.

Reply:
I'm not sure I get what you mean. The `lazy` field in the response is logged as I included the whole `rolloverResponse` in the log.

Reply:
I see but you do not know what the request was.

If you add earlier, at the caller of this method, a variable `boolean lazy = true` and use it in the rollover request and then here too, you would be able to log if a lazy rollover request failed or a "classic" one. Someone reading this log line cannot figure this out without looking at the code.

Reply:
Yes, the code here changed too and we missed the impact. Also there are other maintainers that do not know that here the code always sends a lazy rollover, so it can really help them debug it.

Another option would be to only check the lazy rollover flag. As you said the code is not going to change so why would you keep a generic message here?

Reply:
I agree that this message isn't super valuable in general. Tbh, I'm also fine with dropping this method/log altogether. I can't think of a situation where this log would ever fire -- unless someone accidentally makes the rollover requests dry runs, but tests should cover that, production code should not have to validate that IMO. How do you feel about dropping the method altogether?

Reply:
That is true, no strong opinion here. we could make it an assertion, to indicate that we expect this to be always `true`. What do you think?

Reply:
I think that's a great compromise :)